cmd/spire-agent/cli/run/run.go

```diff
@@ -214,7 +214,8 @@ func newAgentConfig(c *config) (*agent.Config, error) {
 		return nil, err
 	}
 
-	ac.ServerAddress = net.JoinHostPort(c.Agent.ServerAddress, strconv.Itoa(c.Agent.ServerPort))
+	serverHostPort := net.JoinHostPort(c.Agent.ServerAddress, strconv.Itoa(c.Agent.ServerPort))
+	ac.ServerAddress = fmt.Sprintf("dns:///%s", serverHostPort)
```

Comment:
For posterity, from offline conversation

> Will gRPC do the right thing if an IP address is provided with the scheme set to DNS as it is here?

Yes, it will.

https://github.com/grpc/grpc-go/blob/ee21c923a2d1b8c5aa62dea4ce93c2fecac5e687/resolver/dns/dns_resolver.go#L113

> Will gRPC periodically re-resolve this address?

Yes.

https://github.com/grpc/grpc-go/blob/ee21c923a2d1b8c5aa62dea4ce93c2fecac5e687/resolver/dns/dns_resolver.go#L105